### PR TITLE
Added workflow to update current-version-tag in readme.md

### PR DESCRIPTION
**Pull request by thebadcoder96**
Related to Issue #972 

I have changed the hyperlink URL in the readme file and also added a new workflow that will update the link to the live version of the package.

### CODE REVIEW

**Review by taylorfturner — COMMENTED** [review]
Only other comment is validation prior to merge to `main`... maybe change the workflow .yml file to your fork's branch for the time being and just validate

File: .github/workflows/update-current-version.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    - uses: actions/checkout@v4
+
+    - name: Update README.md
+      run: |
+        latest_tag=$(git describe --tags `git rev-list --tags --max-count=1`)
+        sed "s|<current_version_tag>|$latest_tag|g" readme.md > tmp.md && mv tmp.md readme.md
+        git config --local user.email "action@github.com"
+        git config --local user.name "GitHub Action"
+        git commit -am "Update README with latest version tag"
+        git push
```
[the remark this answers is not in the document]
Comment (taylorfturner) [reply .github/workflows/update-current-version.yml:24]: pre-commit hooks failed due to now extra line EOF 
```
Fix End of Files.........................................................Failed
- hook id: end-of-file-fixer
- exit code: 1
- files were modified by this hook

Fixing .github/workflows/update-current-version.yml
```

File: README.md [hunk 2]
```diff
@@ -418,7 +418,7 @@ json_data = Data('https://github.com/capitalone/DataProfiler/blob/main/dataprofi
 ```
 
 If the file type is not automatically identified (rare), you can specify them
-specifically, see section [Specifying a Filetype or Delimiter](#specifying-a-filetype-or-delimiter).
+specifically, see section [Specifying a Filetype or Delimiter](https://capitalone.github.io/DataProfiler/docs/<current_version_tag>/html/profiler.html?#specifying-a-filetype-or-delimiter).
```
[the remark this answers is not in the document]
Comment (thebadcoder96) [reply README.md]: Committed suggested change.

File: .github/workflows/update-current-version.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (thebadcoder96) [reply .github/workflows/update-current-version.yml]: aahhhh...I added the new line EOF as you mentioned.

[the remark this answers is not in the document]
Comment (taylorfturner) [reply .github/workflows/update-current-version.yml]: Perfect! Looks good now on that front -- thx

**Comment by thebadcoder96** [comment]
I am having issues with validating. I changed the .yml to my branch but I am getting an error `fatal: No names found, cannot describe anything.`

Quick googling leads me to believe that this usually means no tags are found. I am a bit confused about how tags work to understand why this is happening. 

I read that this might have something to do with the git clone being shallow. Possible solution could be to add this to the .yml file before the git describe --tags line.

`- run: git fetch --prune --unshallow`

 I have no idea what that means. Please advise

**Comment by thebadcoder96** [comment]
I updated the .yml file to fix the issue I mentioned earlier. And it passed the validation; the github `action-user` committed to change the readme file to my branches tag. 

Please review.

File: README.md [hunk 3]
```diff
@@ -418,7 +418,7 @@ json_data = Data('https://github.com/capitalone/DataProfiler/blob/main/dataprofi
 ```
 
 If the file type is not automatically identified (rare), you can specify them
-specifically, see section [Specifying a Filetype or Delimiter](https://capitalone.github.io/DataProfiler/docs/<workflow_to_replace_with_live_version_tag>/html/profiler.html?#specifying-a-filetype-or-delimiter).
+specifically, see section [Specifying a Filetype or Delimiter](https://capitalone.github.io/DataProfiler/docs/v0.1.0/html/profiler.html?#specifying-a-filetype-or-delimiter).
```
[the remark this answers is not in the document]
Comment (taylorfturner) [reply README.md]: actually this is pulling the first version not 0.10.7...

[the remark this answers is not in the document]
Comment (thebadcoder96) [reply README.md]: oh no, I was just messing around with tags and I gave my branch a new tag called `v0.1.0`. There is no tag like that in the original repo.

**Comment by thebadcoder96** [comment]
The current logic I have used is pulling the latest tag from the latest commit across all branches which is not what we want. We want it to be the latest tag from the latest commit within one branch. I have updated the .yml for that and also tested it on my forked `dev` and `dynamiclink` branches. I also have updated the GitHub actions commit username to my email so it is CLA complaint. 

I am closing this PR and will open a new one with the updated changes so it is CLA complaint.

**Pull request closed without merging**